shared/chat/conversation/list/index.desktop.js

```diff
@@ -1,244 +1,269 @@
+        this._keepIdxVisible = nextProps.messageKeys.indexOf(toFind)
+      }
     }
+  }
 
-    if (this.state.messages !== prevState.messages && prevState.messages.count() > 1) {
-      const prependedCount = this.state.messages.indexOf(prevState.messages.first())
-      const headerCount = this.props.headerMessages.count()
-      if (prependedCount !== -1) {
-        // Measure the new items so we can adjust our scrollTop so your position doesn't jump
-        const scrollTop = this.state.scrollTop + _.range(headerCount, headerCount + prependedCount)
-          .map(index => this._cellMeasurer.getRowHeight({index}))
-          .reduce((total, height) => total + height, 0)
-
-        // Disabling eslint as we normally don't want to call setState in a componentDidUpdate in case you infinitely re-render
-        this.setState({scrollTop}) // eslint-disable-line react/no-did-update-set-state
+  _updateBottomLock = (clientHeight: number, scrollHeight: number, scrollTop: number) => {
+    // meaningless otherwise
+    if (clientHeight) {
+      const isLockedToBottom = scrollTop + clientHeight >= scrollHeight - lockedToBottomSlop
+      if (this.state.isLockedToBottom !== isLockedToBottom) {
+        this.setState({isLockedToBottom})
       }
     }
   }
 
-  componentWillReceiveProps (nextProps: Props) {
-    if (this.props.selectedConversation !== nextProps.selectedConversation) {
-      this.setState({isLockedToBottom: true})
-      this._recomputeList()
+  _maybeLoadMoreMessages = debounce((clientHeight: number, scrollTop: number) => {
+    if (clientHeight && scrollTop === 0) {
+      this.props.onLoadMoreMessages()
     }
+  }, 500)
 
-    const willScrollDown = nextProps.listScrollDownCounter !== this.props.listScrollDownCounter
+  _onScroll = ({clientHeight, scrollHeight, scrollTop}) => {
+    this._scrollTop = scrollTop
+    this._updateBottomLock(clientHeight, scrollHeight, scrollTop)
+    this._maybeLoadMoreMessages(clientHeight, scrollTop)
+  }
 
-    if (!this.state.isScrolling || willScrollDown || this.state.isLockedToBottom) {
-      this._updateInternalMessages(nextProps)
+  _onResize = ({width}) => {
+    if (this._cellCache.columnWidth({index: 0}) !== width) {
+      this._cellCache.clearAll()
     }
+  }
 
-    if (willScrollDown) {
-      this.setState({isLockedToBottom: true})
-    }
+  _rowRenderer = ({index, isScrolling, isVisible, key, parent, style}) => {
+    const messageKey = this.props.messageKeys.get(index)
+    const prevMessageKey = this.props.messageKeys.get(index - 1)
+    const isSelected = messageKey === this.state.selectedMessageKey
+    return (
+      <Virtualized.CellMeasurer
+        cache={this._cellCache}
+        columnIndex={0}
+        key={key}
+        parent={parent}
+        rowIndex={index}>
+        {({measure}) => {
+          const message = messageFactory(messageKey, prevMessageKey, this._onAction, this._onShowEditor, isSelected, measure)
+          return (
+            <div style={style}>
```

Comment:
the handling of the style in the list is handled by this div instead of the inner component. stops them from render thrashing due to the style being merged